apps/api/src/domain/models/digital-text/commands/add-cover-photograph-for-digital-text/add-cover-photograph-for-digital-text.command.integration.spec.ts

```diff
@@ -0,0 +1,186 @@
+                        .create(existingCoverPhotograph);
+                },
+                buildValidCommandFSA: () => fsaFactory.build(),
+            });
+        });
+    });
+
+    describe(`when the command is invalid`, () => {
+        describe(`when the digital text does not exist`, () => {
+            it(`should fail with the expected error`, async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    seedInitialState: async () => {
+                        await testRepositoryProvider
+                            .forResource(ResourceType.photograph)
+                            .create(existingCoverPhotograph);
+                    },
+                    buildCommandFSA: () => fsaFactory.build(),
+                    checkError: (result) => {
+                        assertErrorAsExpected(
+                            result,
+                            new CommandExecutionError([
+                                new AggregateNotFoundError(digitalTextCompositeIdentifier),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+
+        describe(`when the cover photograph does not exist`, () => {
+            it(`should fail with the expected error`, async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    seedInitialState: async () => {
+                        await app
+                            .get(ArangoEventRepository)
+                            .appendEvents(validEventHistoryForDigitalTextWithNoCoverPhotograph);
+                    },
+                    buildCommandFSA: () => fsaFactory.build(),
+                    checkError: (result) => {
+                        assertErrorAsExpected(
+                            result,
+                            new CommandExecutionError([
+                                new InvalidExternalReferenceByAggregateError(
+                                    digitalTextCompositeIdentifier,
+                                    [existingCoverPhotograph.getCompositeIdentifier()]
+                                ),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+
+        describe(`when the digital text already has a cover photograph`, () => {
+            it(`should fail with the expected error`, async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    seedInitialState: async () => {
+                        await app
+                            .get(ArangoEventRepository)
+                            .appendEvents(digitalTextCreated.as(digitalTextCompositeIdentifier));
```

Comment:
You need to have a
```ts
const coverPhotographAddedForDigitalText = digitalTextCreated.andThen<CoverPhotographAddedForDigitalText>(...)
```
and use this test event stream instead. This is what will make things break.

Reply:
You have now done this.